java-util/src/main/java/io/druid/java/util/metrics/cgroups/Memory.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.java.util.metrics.cgroups;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.primitives.Longs;
+import io.druid.java.util.common.logger.Logger;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+public class Memory
+{
+  private static final Logger LOG = new Logger(Memory.class);
+  private static final String CGROUP = "memory";
+  private static final String CGROUP_MEMORY_FILE = "memory.stat";
+  private static final String CGROUP_MEMORY_NUMA_FILE = "memory.numa_stat";
+  private final CgroupDiscoverer cgroupDiscoverer;
+
+  public Memory(CgroupDiscoverer cgroupDiscoverer)
+  {
+    this.cgroupDiscoverer = cgroupDiscoverer;
+  }
+
+  public MemoryStat snapshot()
+  {
+    final MemoryStat memoryStat = new MemoryStat();
+
+    try (final BufferedReader reader = Files.newBufferedReader(
+        Paths.get(cgroupDiscoverer.discover(CGROUP).toString(), CGROUP_MEMORY_FILE)
+    )) {
+      for (String line = reader.readLine(); line != null; line = reader.readLine()) {
+        final String[] parts = line.split(Pattern.quote(" "));
+        if (parts.length != 2) {
+          // ignore
+          break;
```

Comment:
Does this, or the other `break` case just below, happen in the wild? Is it noteworthy if it does? Why ignore the rest of the file instead of just the line with `continue`?

I haven't been able to find an example of anything other than what is the 'expected' input searching a bit online, but docs at https://www.kernel.org/doc/Documentation/cgroup-v1/memory.txt on memory.stat are a bit light other than what the contents are.

Reply:
This is a safety measure, not encountered